profile/src/main/java/com/scottlogic/deg/profile/reader/atomic/AtomicConstraintValueReader.java

```diff
@@ -58,13 +58,17 @@ private DistributedSet getSet(Collection<Object> values, Types type) {
     }
 
     private Object getValue(Object value, Types type) {
-        if (value instanceof Map){
-            return getDate((Map) value);
+        if (type != null) {
+            switch (type) {
+                case NUMERIC:
+                    return getBigDecimal(value);
+                case DATETIME:
+                    if (!(value instanceof String) || value.equals("datetime") || value.equals("years") || value.equals("months")
+                        || value.equals("days") || value.equals("hours") || value.equals("minutes")
+                        || value.equals("seconds") || value.equals("millis")) return value;
+                    return ConstraintReaderHelpers.parseDate((String) value);
```

Comment:
This bit is a little confusing to me. Perhaps some thing like:

```
    private Object getValue(Object value, Types type) {
        if (type == null) {
            return value;
        }
        switch (type) {
            case NUMERIC:
                return getBigDecimal(value);
            case DATETIME:
                if (!(value instanceof String) || Timescale.isTimeScale(value)) {
                    return value;
                }
                return ConstraintReaderHelpers.parseDate((String) value);
            default:
                return value;
        }
    }
```

With the method
```
    public static boolean isTimeScale(Object value) {
        return Stream.of(Timescale.values()).map(Timescale::toString).anyMatch(value::equals);
    }
```
Added in Timescale would be easier to understand. 
Or maybe just break up the if onto seperate lines.